client/Bootstrap/Navigation/useRouteConfig/useRouteConfig.hook.ts

```diff
@@ -18,44 +22,61 @@ const removeWhitespace = (name) => name.replace(/\s/g, '');
 
 const removePathVars = (name) => name.replace(/\${[^}]*\}/g, '');
 
-const contextReducer = (acc, context, page, leaves) => {
-  // TRANSLATE THIS ONCE 188N IS IN
-  const translatedName = context.name;
+const checkFeatureFlag = (page, context, featureFlags) => {
+  if (get(featureFlags.client, context.feature_flag)) {
+    return page.contentComponent;
+  } else {
+    return Error404;
+  }
+};
+
+const contextReducer = (
+  acc,
+  context,
+  page,
+  leaves,
+  featureFlags,
+  translate
+) => {
+  const translatedName = translate(context.name);
   const linkKey = generateKey(translatedName, 'link', context.properties);
   const routeKey = generateKey(translatedName, 'route', context.properties);
+  const component = checkFeatureFlag(page, context, featureFlags);
 
-  if (isTopLevelNav(context.path)) {
-    acc.links.push({
-      to: context.path,
-      key: linkKey,
-      children: translatedName,
-    });
-  } else {
-    leaves.push({
-      parent: `/${extractTopLevelPath(context.path)}`,
-      child: {
-        path: context.path,
-        name: translatedName,
-      },
-    });
+  if (component === page.contentComponent) {
+    if (isTopLevelNav(context.path)) {
+      acc.links.push({
+        to: context.path,
+        key: linkKey,
+        children: translatedName,
+      });
+    } else {
+      leaves.push({
```

Comment:
Why can you not add leaves into the meta at the time? Instead of having to pass in `leaves` to populate and reduce later? It doesn't feel like it's a true reducer.

Reply:
So this was because there's no guarantee that a parent existed due to page order (`/topics/:name/edit` coming before `/topics`). However, I suppose this could check if a parent is there, and if not, create one in preparation.